distributed/worker_memory.py

```diff
@@ -401,7 +401,15 @@ def parse_memory_limit(
     assert isinstance(memory_limit, int)
     if memory_limit == 0:
         return None
-    return min(memory_limit, system.MEMORY_LIMIT)
+    if system.MEMORY_LIMIT < memory_limit:
+        logger.warning(
+            "Ignoring provided memory limit %s due to system memory limit of %s",
+            orig,
+            system.MEMORY_LIMIT,
```

Comment:
I think wrapping these both in `format_bytes` would make this warning more readable